backend/dal/Services/Concrete/ProjectReportService.cs

```diff
@@ -179,10 +179,13 @@ public ProjectReport Add(ProjectReport report)
         public ProjectReport Update(ProjectReport report)
         {
             this.User.ThrowIfNotAuthorized(Permissions.ReportsSpl);
+            var isSPLAdmin = this.User.HasPermission(Permissions.ReportsSplAdmin);
 
             var originalReport = this.Context.ProjectReports
                 .SingleOrDefault(p => p.Id == report.Id) ?? throw new KeyNotFoundException();
-
+            
+            if (!originalReport.IsFinal && report.IsFinal && !isSPLAdmin) throw new InvalidOperationException($"You do not have permissions to make a report final.");
```

Comment:
This exceptions should actually be `NotAuthorizedException` so that they get a 403 instead of a 400.